lib/connection-parameters.js

```diff
@@ -41,14 +40,14 @@ var useSsl = function () {
   return defaults.ssl
 }
 
-var ConnectionParameters = function (config) {
+var ConnectionParameters = function (params) {
   // if a string is passed, it is a raw connection string so we parse it into a config
-  config = typeof config === 'string' ? parse(config) : (config || {})
+  var cfg = typeof params === 'string' ? parse(params) : params || {}
+
   // if the config has a connectionString defined, parse IT into the config we use
   // this will override other default values with what is stored in connectionString
-  if (config.connectionString) {
-    config = parse(config.connectionString)
-  }
+  var config = typeof cfg.connectionString === 'string' ? Object.assign({}, cfg, parse(cfg.connectionString)) : cfg
+
```

Comment:
I’d still suggest the same snippet as earlier, since it always copies the configuration object.